core/src/main/scala/chisel3/Layer.scala

```diff
@@ -63,6 +43,22 @@ object layer {
       case _          => s"${parent.fullName}.$name"
     }
 
+    @tailrec
+    final private[chisel3] def canWriteTo(that: Layer): Boolean = that match {
+      case null              => false
+      case _ if this == that => true
+      case _                 => this.canWriteTo(that.parent)
+    }
+  }
+
+  abstract class BindLayer(
+    val outputDirBehavior: OutputDirBehavior = DefaultOutputDir
+  )(
```

Comment:
Nit: consider just switching to `ExtractLayer` as opposed to `BindLayer` to make this more generic terminology-wise. We've also considered doing this for the "bind" convention in FIRRTL as well.